lib/sqlalchemy/dialects/postgresql/base.py

```diff
@@ -3488,12 +3488,15 @@ def get_check_constraints(self, connection, table_name, schema=None, **kw):
             # "CHECK (((a = 1) OR ((a > 2) AND (a < 5))))"
             # "CHECK (((a > 1) AND (a < 5))) NOT VALID"
             # "CHECK (some_boolean_function(a))"
-            m = re.match(r"^CHECK *\((.+)\)( NOT VALID)?$", src)
+            # "CHECK (((a\n < 1)\n OR\n (a\n >= 5))\n)"
+
+            m = re.match(r"^CHECK *\((.+)\)( NOT VALID)?$", src, flags=re.DOTALL)
             if not m:
                 util.warn("Could not parse CHECK constraint text: %r" % src)
                 sqltext = ""
             else:
-                sqltext = re.sub(r"^\((.+)\)$", r"\1", m.group(1))
+                match_without_newlines = ''.join(m.group(1).splitlines())
```

Comment:
🤦‍♂ Clearly I was a bit too hasty writing this ^_^

I changed this code to instead replace each "line boundary" supported by splitlines (listed here: https://docs.python.org/3.6/library/stdtypes.html#str.splitlines) with a single space, and strip() -ed the result so that a case like:

```
(
a is not null
)
```

will first get matched to ` a is not null ` and then will get stripped to `a is not null`.

I updated the test case I wrote a bit too, since now there's a slight difference from before (since EVERY line separator now gets replaced by its own space). Hope that's ok!